### PR TITLE
Migrate all the CI to GitHub actions

### LINKED ISSUE #2: Get CI working on this repo
I am not able to get travis-ci.org working on this repo. I recommend we use this as a chance to migrate all the CI to github actions.

### PR DESCRIPTION
**Pull request by andersy005**
Fixes #2 

@rabernat, I am curious... Is `pangeo-data` going to be the new home for `xESMF`? I am asking because I haven't updated the badges in the README.rst file yet.

### CODE REVIEW

**Comment by rabernat** [comment]
> Is `pangeo-data` going to be the new home for `xESMF`?

Yes. This is what was decided in https://github.com/JiaweiZhuang/xESMF/issues/98.

File: .github/workflows/pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools setuptools-scm wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: |
+          python setup.py sdist bdist_wheel
+          twine upload dist/*
```
Comment (andersy005) [inline .github/workflows/pypi.yaml:26]: I added this GitHub action that will automatically publish the package on PyPI. @rabernat & @JiaweiZhuang, I am happy to help out with maintaining xesmf including PyPI and conda-forge releases. Feel free to add me on PyPI: https://pypi.org/manage/project/xesmf/collaboration/ as a collaborator. My username is andersy005

**Comment by andersy005** [comment]
This is pretty much ready! I plan to add a few things (such as code style and formatting via black and pre-commit hooks) in separate PR.

File: .github/workflows/pypi.yaml (hunk 1 above)
Comment (andersy005) [reply .github/workflows/pypi.yaml:26]: I currently don't have write permissions on this repo. So, someone with the right permissions both on PyPI and GitHub will need to set up the PYPI_USERNAME and PYPI_PASSWORD secrets.

**Comment by andersy005** [comment]
@pangeo-data/xesmf, any objections to merging this?

**Comment by raphaeldussin** [comment]
@andersy005 LGTM +1! thanks for putting that in! This will be super useful with all the new PRs coming

**Comment by andersy005** [comment]
Thank you @raphaeldussin, @huard! I am going to merge this.

**Comment by raphaeldussin** [comment]
do we need to rebase the PRs for the testing to kick in?

**Pull request merged**